doc/StructuredTypes.md

```diff
@@ -0,0 +1,187 @@
+- by properties names
+- by the constructor.
+
+### Creating objects by properties order
+
+Creating objects by properties order is a default construction method.
+Objects are created by the non-parametrized constructor, and then the n-th Snowflake object field is converted into the n-th customer object property, one by one.
+
+You can annotate your class with `SnowflakeObject` annotation to make sure this creation method would be chosen (however it is not necessary since it is a default method):
+```csharp
+[SnowflakeObject(ConstructionMethod = SnowflakeObjectConstructionMethod.PROPERTIES_ORDER)]
+public class Address
+{
+    public string city { get; set; }
+    public string state { get; set; }
+    public Zip zip { get; set; }
+}
+```
+
+If you would like to skip any customer property, you could use a `[SnowflakeColumn(IgnoreForPropertyOrder = true)]` annotation for a given property.
+For instance, the annotation used in the following class definition makes the `city` be skipped when mapping the properties:
+```csharp
+public class Address
+{
+    [SnowflakeColumn(IgnoreForPropertyOrder = true)]
+    public string city { get; set; }
+    public string state { get; set; }
+    public Zip zip { get; set; }
+}
+```
+
+So, the first field from the database object would be mapped to the `state` property because `city` is skipped.
+
+### Creating objects by property names
+
+Using the `[SnowflakeObject(ConstructionMethod = SnowflakeObjectConstructionMethod.PROPERTIES_NAMES)]` annotation on the customer class can enable the creation of objects by their property names.
+In this creation method, objects are created by the non-parametrised constructor, and then for each of the database object fields a property of the same name is set with the field value.
+It is crucial that database object field names are the same as customer property names; otherwise, a given database object field value would not be set in the customer object.
+You can use the annotation `SnowflakeColumn` to rename the customer object property to the match database object field name.
+
+In the example:
+
+```csharp
+[SnowflakeObject(ConstructionMethod = SnowflakeObjectConstructionMethod.PROPERTIES_NAMES)]
+public class Address
+{
+    [SnowflakeColumn(Name = "nearestCity")]
+    public string city { get; set; }
+    public string state { get; set; }
+    public Zip zip { get; set; }
+}
+```
+
+the database object field `nearestCity` would be mapped to the `city` property of `Adrress` class.
+
+### Creating objects by the constructor
+
+Using the `[SnowflakeObject(ConstructionMethod = SnowflakeObjectConstructionMethod.CONSTRUCTOR)]` annotation on the customer class enables the creation of objects by property names.
+In this creation method, an object with all its fields is created by a constructor.
+A constructor with the exact number of parameters as the number of database object fields should exist because such a constructor would be chosen to instantiate a customer object.
```

Comment:
Does the order of the fields in the constructor matter?

Reply:
yes, mapping is based on fields order. I added a sentence for that.